dts/bindings/sram/intel,sram0.yaml

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+description: Intel ADSP SRAM settings.
+
+compatible: "intel,sram0"
+
+include: base.yaml
+
+properties:
+    reg:
+      required: true
+
+    need-init:
```

Comment:
This looks more like Kconfig material and probably should not be in DTS.

Reply:
@nashif and the others do still look like more DTS? What is about this one that makes it more Kconfig-ish? After all, it's still a "property" from the hardware - I want to better tune my criteria here.

Reply:
The usage in the existing code and here is a bit cumbersome, i.e. if you have the sram already configured in the system and enable, we probably do not need this property, if it is enabled, it will be initialized, platform that do not have this, will not initialize it.
Not only this one, but also `    need-power-on-cache`, I commented on by forgot to push.